File: src/backend/src/main/java/com/personal/finance/backend/bills/service/RecurringBillService.java
package com.personal.finance.backend.bills.service;

import com.personal.finance.backend.bills.dto.request.CreateRecurringBillRequest;
import com.personal.finance.backend.bills.dto.response.RecurringBillDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface RecurringBillService {
    RecurringBillDTO createBill(Long userId, CreateRecurringBillRequest request);
    Page<RecurringBillDTO> getMyBills(Long userId, Pageable pageable);
    void deleteBill(Long id, Long userId);
}

File: src/frontend/src/services/recurringBillService.ts
import api from './api';

const recurringBillService = {
  getBills: async (params?: any) => {
    const response = await api.get('/recurring-bills', { params });
    return response.data;
  },
  createBill: async (data: any) => {
    const response = await api.post('/recurring-bills', data);
    return response.data;
  },
  deleteBill: async (id: number) => {
    const response = await api.delete(`/recurring-bills/${id}`);
    return response.data;
  }
};

export default recurringBillService;